metadata-ws/src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/entities/Study.java

```diff
@@ -84,9 +86,15 @@
     @Column(nullable = false)
     private LocalDate releaseDate;
 
-    @ApiModelProperty(position = 7, dataType = "java.lang.String", notes = "Url to a Taxonomy")
+    @ApiModelProperty(position = 7, dataType = "java.lang.String", notes = "URL to a Taxonomy")
     @JsonProperty
-    @ManyToOne(optional = false)
+    @ManyToOne
+    @JoinFormula(value = "(SELECT DISTINCT taxonomy.id FROM taxonomy " +
```

Comment:
Which is the new test case covering this?

Reply:
Instead of implementing new test cases, I modified the two _existing_ tests which previously covered Study—Taxonomy relation, to test for the new logic. Both tests are in the `study.feature` file.

Test 1 is **“Scenario Outline: search various study by taxonomy name and id”.** It creates five taxonomies, three studies, and then links taxonomies to studies **indirectly** through associating Study with an Analysis, Analysis with a ReferenceSequence, and ReferenceSequence with a Taxonomy. Then we query Studies directly by Taxonomy to ensure that taxonomy fetching is working correctly.

Test 2 is **“Scenario Outline: search various public studies”.** It again creates a taxonomy and several studies and them links them **indirectly** through analyses and reference sequences. The taxonomy fetching is then tested in conjunction with other parameters, such as a release date.